public/TrafficStats.php

```diff
@@ -1,13 +1,79 @@
 <?php
 
 
+use Luracast\Restler\RestException;
 use Respect\Data\Collections\Filtered;
 use Routelandia\Entities;
 use Routelandia\Entities\OrderedStation;
 
 class TrafficStats{
 
-    //TODO: Create function to accept a JSON payload/list of tuples for segments
+    //To test this, use
+    //curl -d 'derp' http://localhost:8080/api/trafficstats/test
+    //curl -X POST http://localhost:8080/api/trafficstats -H "Content-Type: application/json" -d '{"startPoint":"[45.44620177127501,-122.78281856328249]" ,"endPoint": "[45.481798761799084,-122.79243160039188]" }'
+    /**
+     * Takes in a JSON object and returns traffic calculations
+     *
+     * Takes in a JSON object nd returns traffic calculations.
+     * NOTE: $request_Data is not a true JSON object but an
+     * internal restler associative array that handles JSON.
+     * Might need to be changed in the future to dump into a
+     * true JSON object.
+     *
+     * @param $request_data
+     * @return array
+     * @throws RestException
+     * @url POST
+     */
+    public function doPOST ($request_data)
+    {
+        if (empty($request_data)) {
+            throw new RestException(412, "JSON object is empty");
+        }
+         // To grab data from $request_data, syntax is
+         // $request_data['startPoint'];
+
+        return array($request_data);
+    }
+
+    /**
+     * Takes in a float coordinate and returns the station object closest to that point.
+     *
+     * Takes in a float coordinate and returns the station object closest to that point.
+     *
+     * @param float $point
+     * @return null|OrderedStation
+     */
+    function getRelatedStation($point){
+        $s = new Stations();
+        $Station = new OrderedStation();
+
+        $Station = $s->getStationfromCoord($point);
+
+        return $Station;
+
+    }
+
+    /**
+     * Converts JSON object coordinates into floats
+     *
+     * Converts JSON object coordinates into floats
+     * NOTE: 90% sure this wont be needed but it should
+     * be kept incase we change how the project is structured again.
+     *
+     * @param String $coord The String containing JSON coords
+     * @return array float The two coords separated into an array
+     */
+    function parseCoord($coord){
+        $coord = trim($coord,"[]");
+        $pieces = explode(",",$coord);
```

Comment:
It looks like you're manually parsing the JSON string we're expecting to get. 

Why not use json_decode to turn it into an object instead? That might be less fragile, and could give us back an error if the JSON is malformed, rather than trusting that whatever comes in is going to trim and explode correctly.